src/cljs/rems/language.cljs

```diff
@@ -2,11 +2,22 @@
   (:require [clojure.string :as str]
             [re-frame.core :as rf :refer [reg-event-fx reg-sub]]))
 
+(def ^:private language-cookie-name "rems-user-preferred-language")
+
+(defn get-language-cookie []
+  (when-let [value (.. js/document -cookie)]
+    (keyword (second (str/split value #"=")))))
+
+(defn- set-language-cookie! [language]
+  (let [year-from-now (.setFullYear (js/Date.) (inc (.getFullYear (js/Date.))))]
+    (set! (.. js/document -cookie)
+          (str language-cookie-name "=" (name language) "; expires=" (.toString year-from-now) "; path=/"))))
```

Comment:
there's no cookie generator available?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie says that the expiry time should be in GMT, so need to use toUTCString() instead of toString(). But either way, this code should not be written manually.

Reply:
I refactored into UTC string though it's unlikely to be different. There is no need to complicate this feature with encoding here.